zerver/views/__init__.py

```diff
@@ -1307,3 +1316,15 @@ def email_unsubscribe(request, type, token):
 
     return render_to_response('zerver/unsubscribe_link_error.html', {},
                               context_instance=RequestContext(request))
+
+@login_required(login_url = settings.HOME_NOT_LOGGED_IN)
+def mobile_redirect(request):
+    user_profile = request.user
+    params = {
+        "api_key": user_profile.api_key,
+        "email": user_profile.email,
+        "domain": settings.EXTERNAL_HOST
+    }
+    msg = "%s:///?%s" % ("zulip", urlencode(params))
```

Comment:
I assume there's a corresponding patch on the mobile app side that sets up Zulip to handle zulip:/// URLs?  Do we want to suggest installing the app here if it's not installed?  

I have some security concerns with using the API key here; it makes it possible for an someone to steal your API key with 5 seconds at your logged-in Zulip session, which we've avoided by making users type a password to get access to their own API keys.  

I think that change in the security model may be fundamental to the flow, since the mobile app needs the API key in order to work.  I think we could address this security issue somewhat with the following approach:
(1) Sending an email notification to the user whenever a new device is linked using this mechanism (probably a good security improvement anyway) with instructions that if this wasn't them, they can deal by changing their API key.
(2) Using a new randomly generated token instead of the user's API key that only useful for doing a mobile login.  This at least would make it more annoying to extract the API key quickly since it's not literally just looking at the URL from a link.

Eventually we're going to want to have a finer-grained API permissions model and a nice page where you can see linked devices and deactivate the ones you're done with.

@lfaraone thoughts on this?